tests/test_core.py

```diff
@@ -1463,3 +1463,15 @@ def test_method_check_numpy():
         ]
     )
     assert_equal(actual, expected)
+
+
+def test_cumsum() -> None:
+    import numpy_groupies as npg
+
+    group_idx = np.array([4, 3, 3, 4, 4, 1, 1, 1, 7, 8, 7, 4, 3, 3, 1, 1])
+    a = np.array([3, 4, 1, 3, 9, 9, 6, 7, 7, 0, 8, 2, 1, 8, 9, 8])
+    b = npg.aggregate(group_idx, a, func="cumsum")
+
+    bb = groupby_reduce(a, group_idx, func="cumsum", engine="numpy")[0]
```

Comment:
Let's add a `groupby_accumulate(a, group_idx, func="sum", engine="numpy)` that only returns a single array.

Reply:
I'm not a fan that groupby_reduce and groupby_accumulate wont have the same typing though.

Reply:
They are very different operations so they have to be. For e.g. `method="map-reduce"` isn't applicable.